src/sanitise-old-messages/src/sanitiseOldMessages.ts

```diff
@@ -0,0 +1,102 @@
+import { add } from "date-fns"
+import type { Client } from "pg"
+import { logger } from "shared"
+import type { ApiClient, AuditLog, AuditLogDynamoGateway, PromiseResult } from "shared-types"
+import { isError } from "shared-types"
+import type { SanitiseOldMessagesConfig } from "./config"
+
+const rescheduleSanitiseCheck = (
+  dynamo: AuditLogDynamoGateway,
+  message: AuditLog,
+  frequency: Duration
+): PromiseResult<void> => {
+  return dynamo.updateSanitiseCheck(message.messageId, add(new Date(), frequency))
+}
+
+const shouldSanitise = async (
+  db: Client,
+  message: AuditLog,
+  ageThreshold: Duration
+): PromiseResult<{ sanitise: boolean; error: boolean }> => {
+  // Only sanitise messages more than 3 months old
+  if (add(new Date(message.receivedDate), ageThreshold) > new Date()) {
+    return { sanitise: false, error: false }
+  }
+
+  try {
+    const archiveTableResult = await db.query(
+      `SELECT 1
+      FROM br7own.archive_error_list
+      WHERE message_id = $1`,
+      [message.messageId]
+    )
+    if (archiveTableResult.rowCount > 0) {
+      return { sanitise: true, error: false }
+    }
+
+    const unarchivedTableResult = await db.query(
+      `SELECT 1
+      FROM br7own.error_list
+      WHERE message_id = $1`,
+      [message.messageId]
+    )
+    if (unarchivedTableResult.rowCount > 0) {
+      return { sanitise: false, error: false }
+    }
+  } catch (e) {
+    logger.error({
+      message: "Unable to check if message is sanitised",
+      error: e,
+      messageId: message.messageId
+    })
+    return { sanitise: false, error: true }
+  }
+
+  return { sanitise: true, error: false }
+}
+
+const fetchOldMessages = (dynamo: AuditLogDynamoGateway, limit: number): PromiseResult<AuditLog[]> =>
+  dynamo.fetchUnsanitised(limit)
+
+export default async (
```

Comment:
Suggestion: I think if we make this a `class`, we can then share the variables (api, dynamo, and db) between functions by injecting these variables to the class.

Reply:
Done 👍